src/graph/graph_hub.c

```diff
@@ -482,3 +486,83 @@ Attribute_ID FindOrAddAttribute
 	return attr_id;
 }
 
+// create index
+Index AddIndex
+(
+	const char *label,   // label/relationship type
+	const char *attr,    // attribute to index
+	GraphEntityType et,  // entity type (node/edge)
+	IndexFieldType t,    // type of index (range/fulltext/vector)
+	SIValue options,     // index options
+	bool log
+) {
+	ASSERT(label != NULL);
+	ASSERT(attr != NULL);
+	ASSERT(et != GETYPE_UNKNOWN);
+	ASSERT(t == INDEX_FLD_FULLTEXT ||
+		   t == INDEX_FLD_RANGE    ||
+		   t == INDEX_FLD_VECTOR);
+
+	GraphContext *gc = QueryCtx_GetGraphCtx();
+
+	//--------------------------------------------------------------------------
+	// make sure schema exists
+	//--------------------------------------------------------------------------
+
+	SchemaType st = (et == GETYPE_NODE) ? SCHEMA_NODE : SCHEMA_EDGE;
+	Schema *s = GraphContext_GetSchema(gc, label, st);
+
+	// schema missing, creating an index will create the schema
+	if(s == NULL) {
+		s = AddSchema(gc, label, st, log);
+	}
+	ASSERT(s != NULL);
+
+	//--------------------------------------------------------------------------
+	// make sure attribute exists
+	//--------------------------------------------------------------------------
+
+	// creating an index will create the attribute
+	Attribute_ID attr_id = FindOrAddAttribute(gc, attr, log);
+
+	//--------------------------------------------------------------------------
+	// create index field
+	//--------------------------------------------------------------------------
+
+	Index idx = NULL;
+	if(t == INDEX_FLD_RANGE) {
```

Comment:
move to switch case

Reply:
I don't believe I can as `INDEX_FLD_RANGE` isn't an enum value but a macro oring 3 different enum values.
`switch case` can't have `INDEX_FLD_RANGE` as one of its cases